src/custom/components/Settings/SettingsMod.tsx

```diff
@@ -127,14 +128,36 @@ export default function SettingsTab({ className, placeholderSlippage, SettingsBu
 
   const theme = useContext(ThemeContext)
 
-  const [expertMode, toggleExpertMode] = useExpertModeManager()
+  const [expertMode, toggleExpertModeAux] = useExpertModeManager()
+  const toggleExpertMode = useCallback(() => {
+    ReactGA.event({
+      category: 'Expert mode',
+      action: expertMode ? 'Disable Expert Mode' : 'Enable Expert Mode',
+    })
+
+    toggleExpertModeAux()
+  }, [toggleExpertModeAux, expertMode])
```

Comment:
Suggestion, to not depend on `expertMode`:

```suggestion
  const toggleExpertMode = useCallback(() => {
    toggleExpertModeAux((isExperMode) => {
        ReactGA.event({
          category: 'Expert mode',
          action: isExpertMode ? 'Disable Expert Mode' : 'Enable Expert Mode',
        })
        return !isExpertMode
    })
  }, [toggleExpertModeAux])
```